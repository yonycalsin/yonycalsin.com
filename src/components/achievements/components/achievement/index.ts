export { default as Achievement } from './achievement'
