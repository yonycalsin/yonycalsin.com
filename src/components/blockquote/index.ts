export { default as Blockquote } from './blockquote'
