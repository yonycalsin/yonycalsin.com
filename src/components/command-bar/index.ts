export * from './command-bar'
export * from './command-bar-start-action'
