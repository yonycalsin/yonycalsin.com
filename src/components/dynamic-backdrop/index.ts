export { default as DynamicBackdrop } from './dynamic-backdrop'
