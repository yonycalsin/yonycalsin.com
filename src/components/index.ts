export * from './achievements'
export * from './command-bar'
export * from './external-anchor'
export * from './float-socials'
export * from './illustrations'
export * from './mdx-components'
export * from './meta'
export * from './motion'
export * from './night-mode-button'
export * from './page-transition'
export * from './recommendations'
export * from './section'
export * from './work-item'
