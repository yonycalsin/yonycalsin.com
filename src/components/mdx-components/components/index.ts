export * from './anchor'
export * from './linked-heading'
export * from './table'
