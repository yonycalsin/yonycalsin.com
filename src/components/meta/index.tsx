import * as React from 'react'
import Head from 'next/head'

type MetaProps = {
  title?: string
  notRobots?: boolean
}

export const Meta = (props: MetaProps) => {
  const { notRobots = false } = props

  const { description, titleTemplate, defaultTitle, twitterUsername, websiteUrl, socialBannerImage } = {
    description:
      'Frontend developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql.',
    titleTemplate: '%s - Yony Calsin',
    defaultTitle: 'Yony Calsin - Frontend Developer',
    twitterUsername: '@yonycalsin',
    websiteUrl: 'https://www.yonycalsin.com',
    socialBannerImage: 'https://avatars.githubusercontent.com/u/58490737?v=4',
  }

  const schemaOrgJSONLD = [
    {
      '@context': 'https://schema.org',
      '@type': 'Organization',
      url: 'https://www.yonycalsin.com',
      name: defaultTitle,
      alternateName: defaultTitle,
    },
  ]

  return (
    <Head>
      <meta name="robots" content={notRobots ? 'noindex, nofollow, noimageindex, indexifembedded' : 'index, follow'} />
      <meta
        name="googlebot"
        content={notRobots ? 'noindex, nofollow, noimageindex, indexifembedded' : 'index, follow'}
      />

      <meta name="description" content={description} />
      <meta name="image" content={socialBannerImage} />

      <script type="application/ld+json">{JSON.stringify(schemaOrgJSONLD)}</script>

      <link rel="icon" href={socialBannerImage} type="image/png" />
      <meta property="og:url" content={websiteUrl} />
      <meta property="og:type" content="website" />
      <meta property="og:title" content={defaultTitle} />
      <meta property="og:description" content={description} />
      <meta property="og:image" content={socialBannerImage} />

      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:creator" content={twitterUsername} />
      <meta name="twitter:title" content={defaultTitle} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={socialBannerImage} />

      <title>{props.title ? titleTemplate.replace('%s', props.title) : 'Yony Calsin - Desarrollador Frontend'}</title>
    </Head>
  )
}
