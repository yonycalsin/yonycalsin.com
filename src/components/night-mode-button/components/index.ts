export * from './moon-icon'
export * from './sun-icon'
