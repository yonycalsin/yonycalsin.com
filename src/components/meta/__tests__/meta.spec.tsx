/* eslint-disable testing-library/no-container, testing-library/no-node-access */
import * as React from 'react'
import { render } from '@testing-library/react'

import type { MetaProps } from 'typings/components'
import { Meta } from 'components'

function setup(props = {} as Partial<MetaProps>) {
  return render(<Meta {...props} />)
}

function HeadMock(props: React.PropsWithChildren<unknown>) {
  const { children } = props

  return <div>{children}</div>
}

jest.mock('next/head', () => HeadMock)

describe('Meta', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders the meta', () => {
    const view = setup()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders the meta with custom title', () => {
    const view = setup({
      title: 'About Me',
    })

    const title = view.container.querySelector('title')

    expect(title).toHaveTextContent('About Me - Yony Calsin')

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders the meta with no-robots', () => {
    const view = setup({
      notRobots: true,
    })

    const robots = view.container.querySelector('meta[name=robots]')

    expect(robots).toBeInTheDocument()

    expect(robots).toHaveAttribute('content', 'noindex, nofollow, noimageindex, indexifembedded')

    const googlebot = view.container.querySelector('meta[name=googlebot]')

    expect(googlebot).toBeInTheDocument()

    expect(googlebot).toHaveAttribute('content', 'noindex, nofollow, noimageindex, indexifembedded')

    expect(view.baseElement).toMatchSnapshot()
  })
})
