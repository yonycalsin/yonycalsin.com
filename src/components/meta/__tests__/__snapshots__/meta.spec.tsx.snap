// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders the meta 1`] = `
<body>
  <div>
    <div>
      <meta
        content="index, follow"
        name="robots"
      />
      <meta
        content="index, follow"
        name="googlebot"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="image"
      />
      <script
        type="application/ld+json"
      >
        [{"@context":"https://schema.org","@type":"Organization","url":"https://www.yonycalsin.com","name":"Yony Calsin - Software Developer","alternateName":"Yony Calsin - Software Developer"}]
      </script>
      <link
        href="https://avatars.githubusercontent.com/u/58490737?v=4"
        rel="icon"
        type="image/png"
      />
      <meta
        content="https://www.yonycalsin.com"
        property="og:url"
      />
      <meta
        content="website"
        property="og:type"
      />
      <meta
        content="Yony Calsin - Software Developer"
        property="og:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        property="og:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        property="og:image"
      />
      <meta
        content="summary_large_image"
        name="twitter:card"
      />
      <meta
        content="@yonycalsin"
        name="twitter:creator"
      />
      <meta
        content="Yony Calsin - Software Developer"
        name="twitter:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="twitter:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="twitter:image"
      />
      <title>
        Yony Calsin - Software Developer
      </title>
    </div>
  </div>
</body>
`;

exports[`Meta renders the meta with custom title 1`] = `
<body>
  <div>
    <div>
      <meta
        content="index, follow"
        name="robots"
      />
      <meta
        content="index, follow"
        name="googlebot"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="image"
      />
      <script
        type="application/ld+json"
      >
        [{"@context":"https://schema.org","@type":"Organization","url":"https://www.yonycalsin.com","name":"Yony Calsin - Software Developer","alternateName":"Yony Calsin - Software Developer"}]
      </script>
      <link
        href="https://avatars.githubusercontent.com/u/58490737?v=4"
        rel="icon"
        type="image/png"
      />
      <meta
        content="https://www.yonycalsin.com"
        property="og:url"
      />
      <meta
        content="website"
        property="og:type"
      />
      <meta
        content="Yony Calsin - Software Developer"
        property="og:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        property="og:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        property="og:image"
      />
      <meta
        content="summary_large_image"
        name="twitter:card"
      />
      <meta
        content="@yonycalsin"
        name="twitter:creator"
      />
      <meta
        content="Yony Calsin - Software Developer"
        name="twitter:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="twitter:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="twitter:image"
      />
      <title>
        About Me - Yony Calsin
      </title>
    </div>
  </div>
</body>
`;

exports[`Meta renders the meta with no-robots 1`] = `
<body>
  <div>
    <div>
      <meta
        content="noindex, nofollow, noimageindex, indexifembedded"
        name="robots"
      />
      <meta
        content="noindex, nofollow, noimageindex, indexifembedded"
        name="googlebot"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="image"
      />
      <script
        type="application/ld+json"
      >
        [{"@context":"https://schema.org","@type":"Organization","url":"https://www.yonycalsin.com","name":"Yony Calsin - Software Developer","alternateName":"Yony Calsin - Software Developer"}]
      </script>
      <link
        href="https://avatars.githubusercontent.com/u/58490737?v=4"
        rel="icon"
        type="image/png"
      />
      <meta
        content="https://www.yonycalsin.com"
        property="og:url"
      />
      <meta
        content="website"
        property="og:type"
      />
      <meta
        content="Yony Calsin - Software Developer"
        property="og:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        property="og:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        property="og:image"
      />
      <meta
        content="summary_large_image"
        name="twitter:card"
      />
      <meta
        content="@yonycalsin"
        name="twitter:creator"
      />
      <meta
        content="Yony Calsin - Software Developer"
        name="twitter:title"
      />
      <meta
        content="Software Developer creating open source projects and writing on modern JavaScript, Node.js, Typescript and Graphql."
        name="twitter:description"
      />
      <meta
        content="https://avatars.githubusercontent.com/u/58490737?v=4"
        name="twitter:image"
      />
      <title>
        Yony Calsin - Software Developer
      </title>
    </div>
  </div>
</body>
`;
